include/RAJA/policy/hip/atomic.hpp

```diff
@@ -260,6 +273,35 @@ using hip_atomicExch_builtin_types = list<
 
 using hip_atomicCAS_builtin_types = hip_atomicCommon_builtin_types;
 
+template <typename T, enable_if_is_none_of<T, hip_atomicLoad_builtin_types>* = nullptr>
+RAJA_INLINE __device__ T hip_atomicLoad(T volatile *acc)
+{
+  return hip_atomic_CAS_oper(acc, [=] __device__(T a) {
+    return a;
+  });
+}
+
+template <typename T, enable_if_is_any_of<T, hip_atomicLoad_builtin_types>* = nullptr>
+RAJA_INLINE __device__ T hip_atomicLoad(T volatile *acc)
+{
+  return __hip_atomic_load((T *)acc, __ATOMIC_RELAXED, __HIP_MEMORY_SCOPE_AGENT);
+}
```

Comment:
I think this is too constrained, we should be able to load a double via unsigned long long.

Reply:
I've checked and it compiles with all builtin arithmetic types (integral and floating point), as well as enums. So I've changed this to be enabled if std::is_arithmetic<T>::value || std::is_enum<T>::value, and then let all other types fall back to the compare and swap.